spec/spec_helper.rb

```diff
@@ -14,24 +14,27 @@
   end
 end
 
+Object.include Spec::Shared::CassetteSecretsHelper
 VCR.configure do |config|
+  # config.default_cassette_options = { :record => :all }
+
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::Vmware::Engine.root, 'spec/vcr_cassettes')
 
-  secrets = Rails.application.secrets
-  config.define_cassette_placeholder(Rails.application.secrets.vmware_infra_defaults[:hostname]) do
-    Rails.application.secrets.vmware_infra[:hostname]
+  config.define_cassette_placeholder(default_vcr_secret_by_key_path(:vmware_infra, :hostname)) do
+    vcr_secret_by_key_path(:vmware_infra, :hostname)
   end
-  config.define_cassette_placeholder(Rails.application.secrets.vmware_cloud_defaults[:host]) do
-    Rails.application.secrets.vmware_cloud[:host]
+  config.define_cassette_placeholder(default_vcr_secret_by_key_path(:vmware_cloud, :host)) do
+    vcr_secret_by_key_path(:vmware_cloud, :host)
   end
   config.define_cassette_placeholder("VMWARE_CLOUD_AUTHORIZATION") do
-    Base64.encode64("#{Rails.application.secrets.vmware_cloud[:userid]}:#{Rails.application.secrets.vmware_cloud[:password]}").chomp
+    Base64.encode64("#{vcr_secret_by_key_path(:vmware_cloud, :userid)}:#{vcr_secret_by_key_path(:vmware_cloud, :password)}").chomp
   end
   config.define_cassette_placeholder("VMWARE_CLOUD_INVALIDAUTHORIZATION") do
-    Base64.encode64("#{Rails.application.secrets.vmware_cloud[:userid]}:invalid").chomp
-  end
-  secrets.vmware_tanzu.each do |key, val|
-    config.define_cassette_placeholder(secrets.vmware_tanzu_defaults[key]) { val }
+    Base64.encode64("#{vcr_secret_by_key_path(:vmware_cloud, :userid)}:invalid").chomp
   end
+
+  config.define_cassette_placeholder(default_vcr_secret_by_key_path(:vmware_tanzu, :hostname)) { vcr_secret_by_key_path(:vmware_tanzu, :hostname) }
```

Comment:
Not for this PR, but I'm wondering if we consolidate this pattern into a helper method...something like

```suggestion
  vcr_define_cassette_placeholder(config, :vmware_tanzu, :hostname)
```